### PR TITLE
#194 add autocompletion to tag-input

### LINKED ISSUE #194: Tag Input: add autocomplete support

### PR DESCRIPTION
**Pull request by jonthans**
resolves #194

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -204,6 +204,16 @@
...
 
+      <div class="hx-section exampleSection">
+        <div class="example">
+          <h3>Autocomplete Picker</h3>
+          <div id="autocomplete-picker"></div>
+          <script>
+            new hx.AutocompletePicker('#autocomplete-picker', ['Dave', 'Jane', 'Bob', 'Dan', 'Steve', 'Alejandro'])
+          </script>
+        </div>
+      </div>
```
Comment (jonthans) [inline demo/index.html]: I thought this was missing on the demo page so I added it. I guess we should do that for other things that are missing as well at some point?

File: demo/index.html [hunk 2]
```diff
@@ -384,6 +394,18 @@
...
+        <div class="example">
+          <h3>Autocompleted Tag Input</h3>
+          <div id="autocomplete-tag-input"></div>
+          <script>
+            new hx.TagInput('#autocomplete-tag-input', {
+              autocompleteData: ['Jane', 'Bob', 'Dan', 'Steve', 'Alejandro']
+            })
+          </script>
+        </div>
+      </div>
```
Comment (jonthans) [inline demo/index.html]: I found it particularly helpful to play around with this while developing, so I'd vote for adding this for every new module/feature

File: modules/autocomplete-picker/main/index.coffee [hunk 3]
```diff
@@ -224,4 +224,4 @@ hx.autocompletePicker = (items, options) ->
   new AutocompletePicker(selection.node(), items, options)
   selection
 
-hx.AutocompletePicker = AutocompletePicker
\ No newline at end of file
+hx.AutocompletePicker = AutocompletePicker
```
Comment (jonthans) [inline modules/autocomplete-picker/main/index.coffee]: These new-lines at the end of a file are IntelliJ's default addition, which I don't think is a bad idea

**Comment by c-frater** [comment]
LGTM 👍

**Pull request merged**